utils/python/CIME/XML/machines.py

```diff
@@ -64,28 +72,23 @@ def probe_machine_name(self):
         """
         machine = None
         nametomatch = socket.gethostname().split(".")[0]
-        nodes = self.get_node("machine")
+        nodes = self.get_nodes("machine")
 
         for node in nodes:
             machtocheck = node.get("MACH")
             logger.debug("machine is " + machtocheck)
-            self.set_machine(machtocheck)
```

Comment:
This set_machine looked like a dangerous and unexpected side-effect of this method.